tests/src/core/testqgsadjacenttiles.cpp

```diff
@@ -0,0 +1,171 @@
+    Date                 : Jan 7  2016
+    Copyright            : (C) 2016 by Marco Hugentobler
+    Email                : marco at sourcepole dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include <QtTest/QtTest>
+#include <QObject>
+
+#include "qgsapplication.h"
+#include "qgsmaplayerregistry.h"
+#include <qgsrenderchecker.h>
+#include "qgsvectorlayer.h"
+#include "qgsvectorlayerrenderer.h"
+
+/** \ingroup UnitTests
+ * This unit test checks if rendering of adjacent tiles (e.g. to render images for tile caches)
+ * does not result in border effects
+ */
+class TestQgsAdjacentTiles : public QObject
+{
+    Q_OBJECT
+  public:
+    TestQgsAdjacentTiles() {}
+    ~TestQgsAdjacentTiles() {}
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+
+    void testFourAdjacentTiles_data();
+    void testFourAdjacentTiles();
+};
+
+void TestQgsAdjacentTiles::initTestCase()
+{
+  QgsApplication::init();
+  QgsApplication::initQgis();
+}
+
+void TestQgsAdjacentTiles::cleanupTestCase()
+{
+  QgsApplication::exitQgis();
+}
+
+void TestQgsAdjacentTiles::testFourAdjacentTiles_data()
+{
+  QTest::addColumn<QStringList>( "bboxList" );
+  QTest::addColumn<QString>( "resultFile" );
+  QTest::addColumn<QString>( "shapeFile" );
+  QTest::addColumn<QString>( "qmlFile" );
+
+  QString shapeFile = TEST_DATA_DIR + QString( "/france_parts.shp" );
+  QString qmlFile = TEST_DATA_DIR + QString( "/adjacent_tiles/line_pattern_30_degree.qml" );
+  QString resultFile = TEST_DATA_DIR + QString( "/adjacent_tiles/testFourAdjacentTiles1_expected.png" );
+
+  QStringList bboxList1;
+  bboxList1 << "-1.5,48,-0.5,49";
+  bboxList1 << "-0.5,48,0.5,49";
+  bboxList1 << "-1.5,47,-0.5,48";
+  bboxList1 << "-0.5,47,0.5,48";
+
+  QTest::newRow( "testFourAdjacentTiles1" ) << bboxList1 << resultFile << shapeFile << qmlFile;
+
+  qmlFile = TEST_DATA_DIR + QString( "/adjacent_tiles/point_pattern_simple_marker.qml" );
+  resultFile = TEST_DATA_DIR + QString( "/adjacent_tiles/testFourAdjacentTiles2_expected.png" );
+
+  QTest::newRow( "testFourAdjacentTiles2" ) << bboxList1 << resultFile << shapeFile << qmlFile;
+
+  shapeFile = TEST_DATA_DIR + QString( "/lines.shp" );
+  qmlFile = TEST_DATA_DIR + QString( "/adjacent_tiles/simple_line_dashed.qml" );
```

Comment:
I think this line style will need to be a lot thicker so that we can test on other platforms without antialiasing breaking the test